doc/introduction-to-xkb.md

```diff
@@ -0,0 +1,200 @@
+# Introduction to XKB {#xkb-intro}
+
+__XKB__ stands for “X Keyboard Extension”. It may refer to either:
+
+- a [protocol](@ref xkb-the-protocol)
+- a [keyboard layout configuration](@ref xkb-the-config)
+- a [text format](@ref xkb-the-text-format)
+
+## XKB the protocol {#xkb-the-protocol}
+
+A __protocol__ for the [X Windows System], that extends the core protocol.
+
+_xkbcommon’s_ API is somehow derived from this API, but has been
+substantially reworked to expose fewer internal details to clients.
+
+_xkbcommon_ does not depend on a particular windows system; for instance
+it is used by the [Wayland] protocol.
+
+_xkbcommon_ provides the <code>[xkbcommon-x11]</code> module to interface
+a client with a X server. Relevant links:
+
+- [The X Window System Protocol][X Protocol]
+- [The X Keyboard Extension: Protocol Specification][XKB Protocol]
+- [xkbcommon-x11]
+
+
+## XKB the keyboard layout configuration {#xkb-the-config}
+
+In order to use [the protocol](@ref xkb-the-protocol), one must first load a
+[keyboard layout]. The layout usually comes from the OS _layout database_,
+which is commonly [xkeyboard-config]. Since layouts may have definitions in
+common, the database actually stores their basic components separately to allow
+maximum composability and coherence. A recipe to compose a layout from its
+components is called a _layout configuration_.
+
+In XKB, there are several ways to define a layout configuration. They all aim to
+produce a [complete keymap]. The following diagram presents an overview.
+Then they are presented hereinafter, ordered from end user to low-level
+implementation.
+
+@dotfile xkb-configuration "XKB Layout configurations"
+<dl>
+  <dt>
+    RMLVO: <u>R</u>ules, <u>M</u>odel, <u>L</u>ayout, <u>V</u>ariant,
```

Comment:
There's actually somewhat of a layer even above the RMLVO, namely the XML files in xkeyboard-config, that are exposed by xkbregistry, and used by end-user UIs to choose the RMLVO.  But you don't necessarily need to mention it here.

Reply:
I added a note.